keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -2180,11 +2134,39 @@ message NodePrepareResourcesResponse {
     // will be called again for those that are missing.
     map<string, NodePrepareResourceResponse> claims = 1;
 }
+
+message NodePrepareResourceResponse {
+    // These are the additional devices that kubelet must
+    // make available via the container runtime. A claim
+    // may have zero or more requests and each request
+    // may have zero or more devices.
+    repeated Device cdi_devices = 1 [(gogoproto.customname) = "CDIDevices"];
+    // If non-empty, preparing the ResourceClaim failed.
+    // cdi_devices is ignored in that case.
+    string error = 2;
+}
+
+message Device {
+    // A single device instance may map to several CDI device IDs.
+    // None is also valid.
+    repeated string cdi_ids = 1;
+
+    // The request in the claim that this device is associated with.
+    string request_name = 2;
+
+    // A single request may be satisfied by allocating several
+    // different devices. This is the index number starting at zero.
+    int index = 3;
```

Comment:
I wonder whether the index really should be here. We can also drop it and say that devices with the same request name get indexed starting at zero in the order in which they appear in the response.

Reply:
Removed (https://github.com/kubernetes/enhancements/pull/4709#discussion_r1634650510).